site/app/controllers/admin/LateController.php

```diff
@@ -0,0 +1,232 @@
+            foreach($fields as &$field) {
+                $field = trim($field);
+            } unset($field);
+            //Each row has three fields
+            if (count($fields) !== 3) {
+                $data = null;
+                return false;
+            }
+            //$fields[0]: Verify student exists in class (check by student user ID)
+            if(count($this->core->getQueries()->getUserById($fields[0])) !== 1){
+                $data = null;
+                return false;
+            }
+            //$fields[1] represents timestamp in the format (MM/DD/YY) for late days
+            //(MM/DD/YYYY), (MM-DD-YY), or (MM-DD-YYYY).
+            if ($type == "late" && !$this->validate_timestamp($fields[1])) {
+                $data = null;
+                return false;
+            }
+            //$fields[1] represents the gradeable id for extensions
+            if ($type == "extension" && !$this->validate_homework($fields[1])) {
+                $data = null;
+                return false;
+            }
+            //$fields[2]: Number of late days must be an integer >= 0
+            if (!ctype_digit($fields[2])) {
+                $data = null;
+                return false;
+            }
+            //Fields information seems okay.  Push fields onto data array.
+            $data[] = $fields;
+        }
+        //Validation successful.
+        return true;
+    }
+
+    function validate_timestamp($timestamp) {
+    //IN:  $timestamp is actually a date string, not a Unix timestamp.
+    //OUT: TRUE when date string conforms to an accetpable pattern
+    //      FALSE otherwise.
+    //PURPOSE: Validate string to (1) be a valid date and (2) conform to specific
+    //         date patterns.
+    //         'm-d-Y' -> mm-dd-yyyy
+    //         'm-d-y' -> mm-dd-yy
+    //         'm/d/Y' -> mm/dd/yyyy
+    //         'm/d/y' -> mm/dd/yy
+
+        //This bizzare/inverted switch-case block actually does work in PHP.
+        //This operates as a form of "white list" of valid patterns.
+        //This checks to ensure a date pattern is acceptable AND the date actually
+        //exists.  e.g. "02-29-2016" is valid, while "06-31-2016" is not.
+        //That is, 2016 is a leap year, but June has only 30 days.
+        $tmp = array(date_create_from_format('m-d-Y', $timestamp),
+                     date_create_from_format('m/d/Y', $timestamp),
+                     date_create_from_format('m-d-y', $timestamp),
+                     date_create_from_format('m/d/y', $timestamp));
+
+        switch (true) {
+        case ($tmp[0] && $tmp[0]->format('m-d-Y') === $timestamp):
+        case ($tmp[1] && $tmp[1]->format('m/d/Y') === $timestamp):
+        case ($tmp[2] && $tmp[2]->format('m-d-y') === $timestamp):
+        case ($tmp[3] && $tmp[3]->format('m/d/y') === $timestamp):
+            return true;
+        default:
+            return false;
+        }
+        return true;
+    }
```

Comment:
move this function to /app/libraries/DateUtils.php and mark ti static and public, use lowerCamelCase